awscli/customizations/commands.py

```diff
@@ -223,7 +228,8 @@ def _build_subcommand_table(self):
             subcommand_name = subcommand['name']
             subcommand_class = subcommand['command_class']
             subcommand_table[subcommand_name] = subcommand_class(self._session)
-        self._session.emit('building-command-table.%s' % self.NAME,
+        name = self.EVENT_NAME or self.NAME
```

Comment:
This is probably the most minimal change to not impact, but I'm wondering if we should just make it even more general and not even require the an override. I'm mainly asking this because if we do solve this without needing an override, we may be able to fix other issues. For example, the `cloudformation deploy` command collides with the `deploy` service command, so you get weird behavior such as this:
```
$ aws cloudformation deploy dsfd

usage: aws [options] <command> <subcommand> [<subcommand> ...] [parameters]
To see help text, you can run:

  aws help
  aws <command> help
  aws <command> <subcommand> help

aws: error: argument subcommand: Invalid choice, valid choices are:

push                                     | register                                
deregister                               | install                                 
uninstall                               
```
where because the event collides with custom code deploy commands the code deploy commands are injected into `cloudformation deploy`'s command table and causes the command to try to parse it as a command.

As a solution, I'm wondering if we can join the command lineage together on a character that is not used in commands such as underscore to use as the name for the event? So for example we would just make the line:
```python
self._session.emit('building-command-table.%s' % '_'.join([c.name for c in self.lineage])
```
This should avoid any potential future collisions. The one downside is that we may have to update some preexisting handler registrations but hopefully should not be too many.

Also, note I'm not suggesting to use `.` as what to join on because that could introduced undesired behavior where trigger handlers that are intended to be used only for the service command are also triggered for operation commands due to how the event hierarchy works.

Thoughts?